tests/UI/pages/BO/design/imageSettings/index.js

```diff
@@ -0,0 +1,195 @@
+      await this.clickAndWaitForNavigation(page, this.filterResetButton);
+    }
+    await this.waitForVisibleSelector(page, this.filterSearchButton, 2000);
+  }
+
+  /**
+   * Reset and get number of image types
+   * @param page
+   * @return {Promise<number>}
+   */
+  async resetAndGetNumberOfLines(page) {
+    await this.resetFilter(page);
+    return this.getNumberOfElementInGrid(page);
+  }
+
+  /**
+   * Filter image types
+   * @param page
+   * @param filterType
+   * @param filterBy
+   * @param value
+   * @return {Promise<void>}
+   */
+  async filterTable(page, filterType, filterBy, value) {
+    switch (filterType) {
+      case 'input':
+        await this.setValue(page, this.filterColumn(filterBy), value.toString());
+        await this.clickAndWaitForNavigation(page, this.filterSearchButton);
+        break;
+
+      case 'select':
+        await Promise.all([
+          page.waitForNavigation({waitUntil: 'networkidle'}),
+          this.selectByVisibleText(page, this.filterColumn(filterBy), value ? 'Yes' : 'No'),
+        ]);
+        break;
+
+      default:
+        throw new Error(`Filter ${filterBy} was not found`);
+    }
+  }
+
+  /* Column methods */
+
+  /**
+   * Get text from column in table
+   * @param page
+   * @param row
+   * @param columnName
+   * @return {Promise<string>}
+   */
+  async getTextColumn(page, row, columnName) {
+    let columnSelector;
+
+    switch (columnName) {
+      case 'id_image_type':
+        columnSelector = this.tableColumnId(row);
+        break;
+
+      case 'name':
+        columnSelector = this.tableColumnName(row);
+        break;
+
+      case 'width':
+        columnSelector = this.tableColumnWidth(row);
+        break;
+
+      case 'height':
+        columnSelector = this.tableColumnHeight(row);
+        break;
+
+      default:
+        throw new Error(`Column ${columnName} was not found`);
+    }
+
+    return this.getTextContent(page, columnSelector);
+  }
+
+  /**
+   * Get image type status for pages: products, categories, manufacturers, suppliers or stores
+   * @param page
+   * @param row
+   * @param columnName
+   * @return {Promise<boolean>}
+   */
+  async getImageTypeStatus(page, row, columnName) {
+    let columnSelector;
+
+    switch (columnName) {
+      case 'products':
+        columnSelector = this.tableColumnProducts(row);
+        break;
+
+      case 'categories':
+        columnSelector = this.tableColumnCategories(row);
+        break;
+
+      case 'manufacturers':
+        columnSelector = this.tableColumnManufacturers(row);
+        break;
+
+      case 'suppliers':
+        columnSelector = this.tableColumnSuppliers(row);
+        break;
+
+      case 'stores':
+        columnSelector = this.tableColumnStores(row);
+        break;
+
+      default:
+        throw new Error(`Column ${columnName} was not found`);
+    }
+
+    return this.elementVisible(page, `${columnSelector}.action-enabled`, 1000);
```

Comment:
Could you use a selector for this ? If one day HTML markup changes we'll have a hard time finding it (not really but it's good practice to not have hardcoded selector stuff outside of class attributes).